### PR TITLE
Add Dockerfiles

### PR DESCRIPTION
**Pull request by WilliamBZA**

### CODE REVIEW

**Comment by SeanFeldman** [comment]
So the idea is to attempt a reading of the connection string from the environment variable first and if not found, failover to the configuration file? 👍

**Comment by WilliamBZA** [comment]
Correct @SeanFeldman

**Comment by WilliamBZA** [comment]
Rebased on the latest `develop` changes

File: src/dockerfile.asb.endpoint [hunk 1]
```diff
@@ -0,0 +1,21 @@
+FROM mcr.microsoft.com/windows/servercore:1803
```
Comment (SeanFeldman) [inline src/dockerfile.asb.endpoint]: Feels like the docket for should be called something among the lines of what it represents, src/dockerfile.asb.endpointoriented. Having just the `endpoint` part in the name reads as if this was an endpoint file. Don't block this, just sharing good I read it first 🙂

File: src/dockerfile.asb.asbs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+FROM mcr.microsoft.com/windows/servercore:1803
+
+WORKDIR /servicecontrol
+
+ADD /ServiceControl.Transports.ASBS/bin/Release/net461 .
```
Comment (kbaley) [inline src/dockerfile.asb.asbs]: We should consider using an intermediate container to do the actual build but I don't think we need to worry about it just yet.

File: src/dockerfile.asb.asbs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ENV "ServiceControl/DBPath"="C:\\Data\\"
+ENV "ServiceControl/LogPath"="C:\\Logs\\"
+
+EXPOSE 33333
+
+SHELL ["powershell"]
+
+ENTRYPOINT ["powershell", "./runcontainer.ps1"]
+
+# docker run --hostname servicecontrol --name sc -e $connection -v c:/data/:c:/data/ -v c:/logs/:c:/logs/ -it servicecontrolasbs
```
Comment (kbaley) [inline src/dockerfile.asb.asbs]: I would remove this

```suggestion
```

File: src/dockerfile.asb.endpoint [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+ENV "ServiceControl/DBPath"="C:\\Data\\"
+ENV "ServiceControl/LogPath"="C:\\Logs\\"
+
+EXPOSE 33333
+
+SHELL ["powershell"]
+
+ENTRYPOINT ["powershell", "./runcontainer.ps1"]
+
+# docker run --hostname servicecontrol --name sc -e $connection -v c:/data/:c:/data/ -v c:/logs/:c:/logs/ -it servicecontrolasbendpoint
```
Comment (kbaley) [inline src/dockerfile.asb.endpoint]: Remove

```suggestion
```

File: src/dockerfile.asb.endpoint (hunk 1 above)
Comment (WilliamBZA) [reply src/dockerfile.asb.endpoint]: Do you have a suggestion that we could use? We need to give the container image a good name too...

I'll merge in the meantime anyway.

File: src/dockerfile.asb.forwarding [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+ENV "ServiceControl/DBPath"="C:\\Data\\"
+ENV "ServiceControl/LogPath"="C:\\Logs\\"
+
+EXPOSE 33333
+
+SHELL ["powershell"]
+
+ENTRYPOINT ["powershell", "./runcontainer.ps1"]
+
+# docker run --hostname servicecontrol --name sc -e $connection -v c:/data/:c:/data/ -v c:/logs/:c:/logs/ -it servicecontrolasbforwarding
```
Comment (kbaley) [inline src/dockerfile.asb.forwarding]:
```suggestion
```

File: src/dockerfile.asq [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+ENV "ServiceControl/DBPath"="C:\\Data\\"
+ENV "ServiceControl/LogPath"="C:\\Logs\\"
+
+EXPOSE 33333
+
+SHELL ["powershell"]
+
+ENTRYPOINT ["powershell", "./runcontainer.ps1"]
+
+# docker run --hostname servicecontrol --name sc -e $connection -v c:/data/:c:/data/ -v c:/logs/:c:/logs/ -it servicecontrolasq
```
Comment (kbaley) [inline src/dockerfile.asq]:
```suggestion
```

File: src/dockerfile.msmq [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+SHELL ["powershell"]
+
+RUN mkdir c:\\MessageStore
+RUN mkdir c:\\TransactionLogStore
+RUN mkdir c:\\MessageLogStore
+
+RUN powershell -Command Enable-WindowsOptionalFeature -Online -FeatureName MSMQ-Server -All
+RUN powershell -Command Set-MsmqQueueManager -MsgStore C:\MessageStore -TransactionLogStore C:\MessageStore -MsgLogStore C:\MessageStore
+
+# docker run --hostname servicecontrol --name sc -v c:/data/:c:/data/ -v c:/logs/:c:/logs/ -v c:/MessageStore/:c:/a/MessageStore/ -it servicecontrolmsmq
```
Comment (kbaley) [inline src/dockerfile.msmq]:
```suggestion
```

File: src/dockerfile.rabbitmq.conventional [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+ENV "ServiceControl/DBPath"="C:\\Data\\"
+ENV "ServiceControl/LogPath"="C:\\Logs\\"
+
+EXPOSE 33333
+
+SHELL ["powershell"]
+
+ENTRYPOINT ["powershell", "./runcontainer.ps1"]
+
+# docker run --hostname servicecontrol --name sc -e $connection -v c:/data/:c:/data/ -v c:/logs/:c:/logs/ -it servicecontrolrabbitconventional
```
Comment (kbaley) [inline src/dockerfile.rabbitmq.conventional]:
```suggestion
```

File: src/dockerfile.rabbitmq.direct [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+ENV "ServiceControl/DBPath"="C:\\Data\\"
+ENV "ServiceControl/LogPath"="C:\\Logs\\"
+
+EXPOSE 33333
+
+SHELL ["powershell"]
+
+ENTRYPOINT ["powershell", "./runcontainer.ps1"]
+
+# docker run --hostname servicecontrol --name sc -e $connection -v c:/data/:c:/data/ -v c:/logs/:c:/logs/ -it servicecontrolrabbitdirect
```
Comment (kbaley) [inline src/dockerfile.rabbitmq.direct]:
```suggestion
```

File: src/dockerfile.sqlserver [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+ENV "ServiceControl/DBPath"="C:\\Data\\"
+ENV "ServiceControl/LogPath"="C:\\Logs\\"
+
+EXPOSE 33333
+
+SHELL ["powershell"]
+
+ENTRYPOINT ["powershell", "./runcontainer.ps1"]
+
+# docker run --hostname servicecontrol --name sc -e $connection -v c:/data/:c:/data/ -v c:/logs/:c:/logs/ -it servicecontrolsqlserver
```
Comment (kbaley) [inline src/dockerfile.sqlserver]:
```suggestion
```

**Pull request merged**